### PR TITLE
[BEAM-2000] run pylint on specific module

### PR DESCRIPTION
**Pull request by ubunatic**
- added module option: any given args to the script will be used as the module for `pep8` and `pylint`
- cleaned up bash syntax: removed square brackets, using more explicit `test` command instead

### CODE REVIEW

**Comment by aaltay** [comment]
R: @sb2nov

File: sdks/python/run_pylint.sh [hunk 1]
```diff
@@ -39,14 +39,19 @@ EXCLUDED_GENERATED_FILES=(
 
 FILES_TO_IGNORE=""
 for file in "${EXCLUDED_GENERATED_FILES[@]}"; do
-  if [[ $FILES_TO_IGNORE ]]; then
-    FILES_TO_IGNORE="$FILES_TO_IGNORE, "
+  if test -z "$FILES_TO_IGNORE"
+  then FILES_TO_IGNORE="$(basename $file)"
```
Comment (sb2nov) [inline sdks/python/run_pylint.sh]: Nit: Indent the `then` & `else` inside the if

File: sdks/python/run_pylint.sh [hunk 2]
```diff
@@ -39,14 +39,19 @@ EXCLUDED_GENERATED_FILES=(
...
-  FILES_TO_IGNORE="$FILES_TO_IGNORE$(basename $file)"
 done
 echo "Skipping lint for generated files: $FILES_TO_IGNORE"
 
-echo "Running pylint:"
-pylint apache_beam --ignore-patterns="$FILES_TO_IGNORE"
-echo "Running pep8:"
-pep8 apache_beam --exclude="$FILES_TO_IGNORE"
+if test $# -gt 0
+then MODULE="$@"
```
Comment (sb2nov) [inline sdks/python/run_pylint.sh]: Nit: Indent the `then` & `else` inside the if

**Comment by sb2nov** [comment]
Thanks, looks good apart from the two nits.

**Comment by sb2nov** [comment]
LGTM, thanks for this 👍 

@aaltay please merge

File: sdks/python/run_pylint.sh (hunk 1 above)
Comment (ubunatic) [inline sdks/python/run_pylint.sh]: OK.
However, I think the non-indented `then` and `else` is more readable in bash in this case, i.e., you should write either:

    if ...
    then single command
    else single command
    fi

or for multiple commands:

    if ...; then
       indented commands
       more commands
    else
       indented command
       more commands
    fi

File: sdks/python/run_pylint.sh (hunk 2 above)
Comment (ubunatic) [inline sdks/python/run_pylint.sh]: Done

**Comment by aaltay** [comment]
Could you add a comment to the top of the file on how to use this (something like ./run_pylint.sh [module], module defaults apache_beam)

**Comment by ubunatic** [comment]
OK I added the usage info.

Other question:
Did I miss any steps to make the checks above succeed?
If they fail for most/all PRs, shouldn't we probably disable or fix them first?

**Comment by aaltay** [comment]
LGTM.

Travis tests are being currently removed. Although Travis tests give valid signals for Python development they are taking too long and timing out for Java tests. Part of the reason they are still there is Python on Mac tests are useful. (@jasonkuster may have more details)

Jenkins tests should not fail. If they do for an unrelated reason (like in this case) please file an issue for a flaky/failing tests.

**Comment by ubunatic** [comment]
Thx for the info. The Jenkins test failed because of the following flaky test https://issues.apache.org/jira/browse/BEAM-1868

**Comment by sb2nov** [comment]
Retest this please

**Pull request closed without merging**